handlers/refunds.go

```diff
@@ -0,0 +1,69 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/companieshouse/chs.go/log"
+	"github.com/companieshouse/payments.api.ch.gov.uk/mappers"
+	"github.com/companieshouse/payments.api.ch.gov.uk/models"
+	"github.com/companieshouse/payments.api.ch.gov.uk/service"
+	"github.com/gorilla/mux"
+	"net/http"
+)
+
+// HandleCreateRefund initiates a refund from the external provider
+func HandleCreateRefund(w http.ResponseWriter, req *http.Request) {
+	if req.Body == nil {
+		log.ErrorR(req, fmt.Errorf("request body empty"))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	vars := mux.Vars(req)
+	id := vars["paymentId"]
+	if id == "" {
+		log.ErrorR(req, fmt.Errorf("payment id not supplied"))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	requestDecoder := json.NewDecoder(req.Body)
+	var incomingRefundResourceRequest models.CreateRefundRequest
+	err := requestDecoder.Decode(&incomingRefundResourceRequest)
+	if err != nil {
+		log.ErrorR(req, fmt.Errorf("request body invalid: [%v]", err))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// once we've read and decoded request body call the refund service handle internal business logic
+	refund, responseType, err := refundService.CreateRefund(req, id, incomingRefundResourceRequest)
+
+	if err != nil {
+		log.ErrorR(req, fmt.Errorf("error creating refund resource: [%v]", err), log.Data{"service_response_type": responseType.String()})
+		switch responseType {
+		case service.InvalidData:
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		case service.Error:
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		default:
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+	}
+
+	refundResource := mappers.MapToRefundResponse(*refund)
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusCreated)
```

Comment:
I agree that 201 is correct here, although would be good to confirm what status code `POST /payments` returns for consistency.

Reply:
Checked, it is consistent